packages/menu/src/MenuItem/InternalMenuItemContent.tsx

```diff
@@ -53,9 +56,13 @@ export const InternalMenuItemContent = React.forwardRef<
     const { as } = useInferredPolymorphic(asProp, rest, 'button');
 
     const { theme, darkMode, highlight, renderDarkMenu } = useMenuContext();
-    const { depth, hasIcon: parentHasIcon } = useSubMenuContext();
-    const isSubMenuItem = depth > 0;
-    const highlighted = id === highlight?.id;
+    const { depth: submenuDepth, hasIcon: submenuHasIcon } =
+      useSubMenuContext();
+    const { depth: groupDepth, hasIcon: groupHasIcon } = useMenuGroupContext();
+    const isNested = !!(submenuDepth || groupDepth);
+    // @ts-expect-error - highlighted isn't a prop on this component, but could be passed in from MenuItem
```

Comment:
Should it be added as a prop?

Reply:
It's not the ideal way to set `highlighted`, since it will cause conflicts with the `HighlightReducer`. I guess I could add the prop and add TSDoc to say "don't use this"

The only reason I have this `forceHighlight` is so we can test in Storybook without needing to create a descendants context

Reply:
Maybe you can add a comment in here that this is for testing in storybook?